src/waypoint.jsx

```diff
@@ -168,65 +168,44 @@ export default class Waypoint extends React.Component {
   }
 
   /**
-   * @param {Object} node
-   * @return {Number}
-   */
-  _distanceToTopOfScrollableAncestor(node) {
-    if (this.scrollableAncestor !== window && !node.offsetParent) {
-      return null;
-    }
-
-    if (this.scrollableAncestor === window) {
-      const rect = node.getBoundingClientRect();
-      return rect.top + window.pageYOffset - document.documentElement.clientTop;
-    }
-
-    if (node.offsetParent === this.scrollableAncestor || !node.offsetParent) {
-      return node.offsetTop;
-    } else {
-      const nextOffset =
-        this._distanceToTopOfScrollableAncestor(node.offsetParent);
-      return nextOffset === null ? null : node.offsetTop + nextOffset;
-    }
-  }
-
-  /**
    * @return {string} The current position of the waypoint in relation to the
    *   visible portion of the scrollable parent. One of `POSITIONS.above`,
    *   `POSITIONS.below`, or `POSITIONS.inside`.
    */
   _currentPosition() {
-    const waypointTop =
-      this._distanceToTopOfScrollableAncestor(ReactDOM.findDOMNode(this));
-    if (waypointTop === null) {
-      // not visible
-      return POSITIONS.invisible;
-    }
```

Comment:
This will behave slightly different now. For example, if you run `document.createElement('div').getBoundingClientRect();` you'll notice that all values are 0, which means that we'll return that it's inside the window (if that's the scrollable ancestor). If we want to keep the previous behaviour, we should add an explicit check for visibility.